site/src/sphinx/client-circuit-breaker.rst

```diff
@@ -1,6 +1,270 @@
+    final CircuitBreaker circuitBreaker = CircuitBreaker.ofDefaultName();
+    final HttpClient client = new ClientBuilder(...)
+            .decorator(HttpRequest.class, HttpResponse.class,
+                       CircuitBreakerHttpClient.newDecorator(circuitBreaker, strategy))
+            .build(HttpClient.class);
+
+    client.execute(...).aggregate().join(); // Send requests on and on.
+
+Now, the :api:`Client` can track the number of success or failure events depending on the :apiplural:`Response`.
+The :api:`CircuitBreaker` will be ``OPEN``, when the number of failures divided by the total number of
+:apiplural:`Request` exceeds the failure rate. Then the :api:`Client` will immediately get
+:api:`FailFastException` by the :api:`CircuitBreaker`.
+
+.. _circuit-breaker-strategy:
+
+``CircuitBreakerStrategy``
+--------------------------
+
+How does the :api:`CircuitBreaker` know whether a :api:`Response` is successful or not?
+:api:`CircuitBreakerStrategy` does the job. In the above example, if a :api:`Response`'s status is ``5xx``
+or an ``Exception`` is raised during the call, the count of failure is increased.
+Of course, you can have your own ``strategy`` by implementing :api:`CircuitBreakerStrategy`.
+
+.. code-block:: java
+
+    import com.linecorp.armeria.common.HttpStatusClass;
+
+    final CircuitBreakerStrategy<HttpResponse> myStrategy = new CircuitBreakerStrategy<HttpResponse>() {
+
+        @Override
+        public CompletableFuture<Boolean> shouldReportAsSuccess(HttpResponse response) {
+            return response.aggregate().handle((res, cause) -> {
+                if (cause != null) { // A failure if an Exception is raised.
+                    return false;
+                }
+
+                final HttpStatus status = res.status();
+                if (status != null) {
+                    // A failure if the response is 5xx.
+                    if (status.codeClass() == HttpStatusClass.SERVER_ERROR) {
+                        return false;
+                    }
+
+                    // A success if the response is 2xx.
+                    if (status.codeClass() == HttpStatusClass.SUCCESS) {
+                        return true;
+                    }
+                }
+
+                // Not a success nor a failure. Do not include this response.
+                return null;
+            });
+        }
+    };
+
+If you want to treat a :api:`Response` as a success, return ``true``. And return ``false`` as a failure.
+Note that :api:`CircuitBreakerStrategy` can return ``null`` as well. It won't be counted as a success nor
+a failure.
+
+``Grouping``
+------------
+
+In the very first example above, a single :api:`CircuitBreaker` was used to track the events. However,
+using only one :api:`CircuitBreaker` is not recommend. There might be an API which needs heavy calculation
+causing failures frequently. On the other hands, there can be another API which does not need resources
```

Comment:
- hands -> hand
- does need resources and simply respond. -> is not resource hungry and thus is not likely to fail.